### ISSUE TITLE
Certain non-alphanumeric characters not always accepted as Abbreviations.

### ISSUE THREAD

**Issue by Elliria** [autokey/autokey#621]
## Classification:
Bug

## Reproducibility:
Sometimes

## Versions:
The environment for testing this in AutoKey 0.90.4 was Ubuntu MATE 16.04 LTS.
The environment for testing this in AutoKey 0.95.10 was Kubuntu 20.04 LTS.

## Used GUI:
Gtk and Qt.

## Installed via:
Package manager.

## Linux Distribution:
Ubuntu MATE 16.04 LTS and Kubuntu 20.04 LTS.

## Summary:
There's an AutoKey behavior that has changed from version 0.09.4 to the current 0.95.10 that may be a bug. This is an intermittent issue that cannot consistently be repeated.

Regardless of which options (checkboxes) you select when creating an Abbreviation for a phrase or script, AutoKey 0.90.4 accepted three dashes as an Abbreviation, but AutoKey 0.95.10 sometimes doesn't.

When AutoKey 0.95.10 has the issue, if you try to set three dashes as the Abbreviation, AutoKey ignores your input entirely and does nothing.

I have not tested other non-alphanumeric characters as Abbreviations, but it's possible some or all of those would behave similarly.

## Steps to Reproduce:
	1. Create a phrase or script.
	2. Press the "Set" button next to "Abbreviations: (None configured)".
	3. Press the "Add" button.
	4. Type three dashes.
	5. Press the "Enter" key.
	6. Press the "OK" button.
	7. Check the "Abbr." column to see if the abbreviation was set.
	8. If not, repeat steps 1 through 7.

## Expected Results:
The Abbreviation you specified should be set.

## Actual Results:
Sometimes no Abbreviation is set and no error message is shown. Your input is simply ignored.

## Notes:
Work-around: If you can't get AutoKey to accept the three dashes as an Abbreviation, repeat steps 1 through 7 above and type x--- instead of three dashes in step 4. Then press the OK button and check for the presence of the Abbreviation. Then edit the Abbreviation and remove the x and press the "OK" button. The three-dash Abbreviation then works without issue.

**Comment by josephj11** [comment]
This should probably work. It may be related to some characters not matching the Python `\w` set of word characters.

Even if it doesn't/shouldn't work, it should complain about it consistently.

**Comment by Elliria** [comment]
Yeah, I hesitated to report it here because I couldn't get it to consistently misbehave, but it does happen some of the time, so it might be of interest.

**Comment by josephj11** [comment]
As I mentioned on Gitter, intermittent bugs are totally valid. They're just harder to analyse and fix which makes reporting them carefully and in detail even more important.

It really helps when there are multiple reports (collected in one place) which can be compared and contrasted. It also helps to see how widespread an issue is, like determining whether it may be isolated to one user/installation or if it affects everyone and in evaluating what priority should be assigned to fixing it.

If something goes wrong more than once, it's a real bug. If it is intermittent, that's just a characteristic of the bug.

The challenge is to be able to recreate the bug so it can be analysed and the more information we have, the better.

E.g.: If two user report the same bug and are using different front ends, then the problem probably isn't in the front end code. Or, if all the reports occur when the same application window is active, then we know to use that application when attempting to recreate the problem and we can consider what may be unique about that application.

**Comment by Elliria** [comment]
Yeah, since it's inconsistent, I wasn't able to verify whether it exists in one front-end and not another. I'm also going to be curious if we get any "me, too" messages in here.

**Comment by BlueDrink9** [comment]
a brilliant thing to do (if you know a little git and want to learn to harness one of its most powerful features) is to **bisect** which version/commit introduced this change. Would be tricky to do unless you can trigger it reliably though

**Comment by Elliria** [comment]
The problem is that I haven't been able to get it to reliably happen. It's just a very occasional thing with no changes that I'm aware of or that I intentionally cause to anything else on the system during the attempts. It's possible there's some rarely-occurring operating-system (or other software) script that happens to fire off in the background and causes all the stars to align to cause the issue and poor little AutoKey is just its innocent victim.
